### DISCUSSION TITLE
MintPy in Debian

### DISCUSSION THREAD

**Discussion by avalentino** [insarlab/MintPy#810] [General]
Hi,
I would like to inform you all that a Debian package for MintPy is now in the official Debian archive.
This means the MintPy, and also pysolid and pyaps, can be installed on Debian with a simple apt command:
```
$ sudo apt install mintpy
```
The package should flow to Ubuntu and other Debian derivatives as new releases come.

Please refer to the [`/usr/share/doc/python3-mintpy/README.Debian`](https://salsa.debian.org/debian-gis-team/mintpy/-/blob/master/debian/README.Debian) file for more details.

Any feedback is very welcome.

**Comment by yunjunz** [comment]
This is awesome @avalentino! Thank you for making it happen!

We should add this to the [installation note](https://github.com/insarlab/MintPy/blob/main/docs/installation.md). It would be helpful to also add a link to the "Debian derivatives" OS names, to help users who are not familiar with Debian (like myself) to decide if this option works for them.

I have a couple of questions:
1. Will the `apt install` works for `pysolid` or `pyaps` alone, e.g., if one wants to install `pysolid` only without `mintpy`?
2. How does the maintenance of the [debian/mintpy](https://salsa.debian.org/debian-gis-team/mintpy) work? Will it be updated for every release made in this GitHub repo, manually or auto-synced like [this](https://github.com/nisar-solid/ATBD/blob/main/.github/workflows/gitlab_sync.yml) using github workflow?

As for the feedback, the instruction you showed is clear and straightforward, I would only suggest maybe using `mintpy_` as prefix instead of `mintpy-`, so that it is more consistent in style with other snake_case script names?

**Reply by avalentino** [reply]
Thanks for the feedback @yunjunz.
ASAP I will provide a small PR to update the [installation notes](https://github.com/insarlab/MintPy/blob/main/docs/installation.md).

Regarding you questions:

1. yes, of course pysolid and pyaps can be installed independently with the same mechanism. Package names are: `python3-pysolid` and `python3-pyaps3`. For MintPy it is possible to install `python3-mintpy`, including only the python package (without command line tools) and the `mintpy`, including only the command line tools. Of course, to satisfy dependencies, when you install `mintpy` the entire stack of packages is installed.
2. the mintpy package in Debian (as well as pysolid and pyaps) are maintained by the [Debian GIS Team](https://wiki.debian.org/DebianGis) (including me). The update, at the moment, will be done manually but the plan is to update at each release as son as it comes out.

Finally, regarding the names of the scripts/entry points, yes I can easily change them to use snake_case style.
By the way I would suggest to change the names directly in the MintPy `setup.py` so that in Debian we do not have do modify anything with respect to the standard installation. If you thin that this is feasible and there are no bug issue with backward compatibility I could provide a patch for that (maybe not immediately because this is a very busy period).

**Reply by yunjunz** [reply]
Thank you for the explanation @avalentino, and for the maintenance.

Could you please also update the installation with the `apt install` on the repo of `pyaps` and `pysolid`?

I agree that changing in mintpy `setup.py` is the way to go. Since this change will affect all the current users with existing installation methods, I prefer to wait for a future time to do this, when we have more compact script names and numbers.

**Reply by avalentino** [reply]
OK, I will update the script prefixes in the debian package ASAP.

Regarding possible future evolutions of the entry points for mintpy, an option could be to provide a single entry point (`mintpy`) with multiple sub-commands. This one could live side by side with the current entry point for a TBD transition period.
If you think the the approach makes sense I could work to a PR in that direction.

> Could you please also update the installation with the apt install on the repo of pyaps and pysolid?

sure, I will do it ASAP.

**Reply by yunjunz** [reply]
Could you explain a little bit more about the single entry point approach? Is that what `gmt` is using? And in that case, what people should do if they want to run non-mintpy commands?

**Reply by avalentino** [reply]
I don't know well `gmt` but yes, the approach seems similar.
The single entry point (the `mintpy` script) will provide access to all functions currently provided bu the set of entry point currently provided via sub-commands:
```sh
$ mintpy -h

USAGE: mintpy [-h][options]

Long desctiption

Subcommands:
  ...
  pre_snap:          description
  smallbaselineApp:  description
  save_qgis:         description
  view:              description
```

To access the specific function:

```sh
$ mintpy smallbaselineApp [-h][options] <arguments>
```

Not sure about what you mean by "non-mintpy commands".
The idea is that, in principle, any function available in the mintpy python package can be (eventually wrapped and) exposed as a sub-command.

**Reply by yunjunz** [reply]
Thank you for the explanation @avalentino. Yes, this single entry point sounds great to me! A PR is more than welcome.

**Comment by avalentino** [comment]
FYI: all the packages are now also available for Ubuntu 22.04 on the eotools PPA: https://launchpad.net/~a.valentino/+archive/ubuntu/eotools?field.series_filter=jammy

**Reply by yunjunz** [reply]
Very cool! Thanks!
